### ISSUE TITLE
z Open Editor is showing red lines under copybooks even though they are there

### ISSUE THREAD

**Issue by oakmount66** [IBM/zopeneditor-about#363]
Sometimes when we open a COBOL program not all copybooks are marked as found, but still have this red line below.
When looking in the log for z Open Editor and local on the PC the copybook is there.
See sample in picture below.
![COBOL copybook error](https://github.com/IBM/zopeneditor-about/assets/60236192/bfb633d5-a895-4e01-be4e-86f2a25199eb)

Se also runtime environment versions in below:
![COBOL copybook error - versions](https://github.com/IBM/zopeneditor-about/assets/60236192/7e05d856-1c9d-4bfb-a5d9-541501305e29)

**Comment by phaumer** [comment]
@oakmount66 I am not able to reproduce this issue. I was testing with 1000 copybooks via z/OSMF on Windows, which is what we now do for our acceptance tests before we ship as well. 

1. In the log output of you screenshot you see the path to the temp directory in which the files get downloaded. If you open such as file directly from the file system in that location: Does it have the right contents? Is the encoding correct? 
2. If it does have the right content: do the red squiggles go away as soon as you start editing the program (i.e. then we reparse the program and perhaps the file will then be read correctly). That could be an indication that the parser had some issues. In that case it would be good to get [the full Debug log as well as the language server log](https://ibm.github.io/zopeneditor-about/Docs/locating_local_client_logs.html#locating-vs-code-extension-and-lsp-log-files-on-the-local-client) for further investigation.

**Comment by oakmount66** [comment]
Hi Peter. 
Regarding #1 yes it's there, it's the right content and the right encoding.
I haven't had the opportunity to test #2 but will get back as soon as possible.

**Comment by oakmount66** [comment]
Hi Peter.
Now I have the logs etc. I'm not keen on putting them into the site here cause some of the files includes proprietary code. Can I in anyway send them to you personally?

**Comment by antonkc** [comment]
Greetings @phaumer,

I have a look-alike problem. I will describe it here for you to judge if I should open a new issue or reporting here is good enough.

After some testing I managed to reliably reproduce through these steps:

1. Open Cobol file with copybooks
2. Open the same cobol file with another name
3. Compare both files ("File: Compare active file with..." command)
4. Close both files, but keep the comparison
5. Close VSCode
6. Open VSCode (Comparison should be restored without opening the files)
7. Review copybook errors. Often, one side reports the lack of copybooks despite them being donwloaded.

Additional observations:
- Editing fixes the errors, sometimes at least.
- Logs report downloading the copybooks. In some instaces multiple times. When they were downloaded only once, the error appears in one side of the comparison.

Additional info:
zowe version - 7.9.1
![VSCode version](https://github.com/IBM/zopeneditor-about/assets/31370603/adafad05-2a2c-4e32-ab4b-ee2e70acd76c)
![IBM Z Open Editor Version 3.3.3](https://github.com/IBM/zopeneditor-about/assets/31370603/83c091cb-ba70-46c0-a1e7-950fa5106a76)
![ZOWE explorer version 2.13.1](https://github.com/IBM/zopeneditor-about/assets/31370603/ebc4aaf1-dcb7-4cbf-ba4e-8bd28c4e125d)

Best regards.

**Comment by sdaimwood** [comment]
> Greetings @phaumer,
> 
> I have a look-alike problem. I will describe it here for you to judge if I should open a new issue or reporting here is good enough.
> 
> After some testing I managed to reliably reproduce through these steps:
> 
> ...


Thank you @antonkc for the detailed steps, I was able to reproduce this inconsistent behavior as well. We will investigate and determine a fix. As you noted, editing the document should result in the links correctly showing and the errors disappearing, but this should be handled correctly from the start.

**Comment by phaumer** [comment]
Fixed in v3.4.0.
